### PR TITLE
User menu

### PR DESCRIPTION
**Pull request by vshovkuta**
**Global changes:**
\+ added defaultUserPic
\+ added defailtShelterPic
\+ added navbar module
\^ updated main conteriner style (padding 5px => 50px)

**Create user-menu component**
\+ added user-menu component
\+ added user-menu module
\> moved "Sign in" button to user-menu component
\^ updated variables.sccs (add $default-userPic-color)
\^ updated authentication service (**_temporary_**)

user-menu

### CODE REVIEW

File: src/app/navbar/user-menu/user-menu.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { UserMenuComponent } from './user-menu.component';
+
+describe('UserMenuComponent', () => {
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.spec.ts]: mark it with x or fix the tests

File: src/app/navbar/user-menu/user-menu.component.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div *ngIf="isLogged(); else guest">
+
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.html]: extra empty line

File: src/app/navbar/user-menu/user-menu.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-user-menu',
+  templateUrl: './user-menu.component.html',
+  styleUrls: ['./user-menu.component.scss']
+})
+export class UserMenuComponent implements OnInit {
+  constructor() { }
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.ts]: do not keep empty constructor and life cycle hooks

File: src/app/navbar/user-menu/user-menu.component.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  public isLogged(): boolean {
+    return Boolean(localStorage.getItem('user'));
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.ts]: do not use localStorage directly, if you need anything on window object you need to create injectable object for it and add it as dependency to your service (yes, service - component is wrong place for such logic)

File: src/app/navbar/user-menu/user-menu.component.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  public getUser(): string | boolean {
+    if (this.isLogged()) {
+      return JSON.parse(localStorage.getItem('user'))[0]['email'];
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.ts]: same here

File: src/app/navbar/user-menu/user-menu.component.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  styleUrls: ['./user-menu.component.scss']
+})
+export class UserMenuComponent implements OnInit {
+  constructor() { }
+
+  public isLogged(): boolean {
+    return Boolean(localStorage.getItem('user'));
+  }
+
+  public getUser(): string | boolean {
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.ts]: you should decide on the type, if should not be string or boolean, you can keep it as just string as keep it as null if you don't have any

File: src/app/navbar/user-menu/user-menu.component.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    if (this.isLogged()) {
+      return JSON.parse(localStorage.getItem('user'))[0]['email'];
+    } else {
+      return false;
+    }
+  }
+
+  public logOut(): void {
+    localStorage.removeItem('user');
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.component.ts]: same here, not the right place to work with localStorage

File: src/app/navbar/user-menu/user-menu.module.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatMenuModule } from '@angular/material/menu';
+import { MatIconModule } from '@angular/material/icon';
+import { UserMenuComponent } from './user-menu.component';
+import { AppRoutingModule } from '../../app-routing.module';
+
+
+
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.module.ts]: keep just 1 empty line

File: src/app/navbar/user-menu/user-menu.module.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+@NgModule({
+  declarations: [UserMenuComponent],
+  imports: [
+    CommonModule,
+    AppRoutingModule,
+    MatMenuModule,
+    MatIconModule,
+    AppRoutingModule
```
Comment (ksenya-ta) [inline src/app/navbar/user-menu/user-menu.module.ts]: why do you need it here? and twice?

File: src/app/shared/services/user/authentication.service.ts [hunk 10]
```diff
@@ -25,6 +25,18 @@ export class AuthenticationService {
...
+    const body: object[] = [{
+      id: '8c338aa3-2e1b-4a44-aab0-4c4045d39e91',
+      phoneNumber: '8045357385',
+      email: loginData['email'].toLowerCase(),
+      password: null,
+      roles: [
+        'Admin'
+      ]
+    }];
+    localStorage.setItem('user', JSON.stringify(body));
```
Comment (ksenya-ta) [inline src/app/shared/services/user/authentication.service.ts]: do not use localStorage directly, inject window and they you can use it from there

File: src/app/navbar/user-menu/user-menu.component.html (hunk 2 above)
Comment (vshovkuta) [reply src/app/navbar/user-menu/user-menu.component.html]: fixed

File: src/app/navbar/user-menu/user-menu.component.ts (hunk 3 above)
Comment (vshovkuta) [reply src/app/navbar/user-menu/user-menu.component.ts]: fixed

File: src/app/navbar/user-menu/user-menu.component.ts (hunk 6 above)
Comment (vshovkuta) [reply src/app/navbar/user-menu/user-menu.component.ts]: fixed

File: src/app/navbar/user-menu/user-menu.module.ts (hunk 8 above)
Comment (vshovkuta) [reply src/app/navbar/user-menu/user-menu.module.ts]: fixed

File: src/app/navbar/user-menu/user-menu.module.ts (hunk 9 above)
Comment (vshovkuta) [reply src/app/navbar/user-menu/user-menu.module.ts]: It was part of another solution... 
fixed

File: src/app/navbar/user-menu/user-menu.component.ts (hunk 3 above)
Comment (ksenya-ta) [reply src/app/navbar/user-menu/user-menu.component.ts]: @vshovkuta not fixed - constructor() { } is still here

File: src/app/navbar/user-menu/user-menu.component.spec.ts (hunk 1 above)
Comment (vshovkuta) [reply src/app/navbar/user-menu/user-menu.component.spec.ts]: fixed

File: src/app/shared/services/user/authentication.service.ts [hunk 11]
```diff
@@ -21,10 +22,22 @@ export class AuthenticationService {
...
+    const body: object[] = [{
+      id: '8c338aa3-2e1b-4a44-aab0-4c4045d39e91',
+      phoneNumber: '8045357385',
+      email: loginData['email'].toLowerCase(),
+      password: null,
+      roles: [
+        'Admin'
+      ]
+    }];
+    LocalStorageService.setItem('user', JSON.stringify(body));
```
Comment (ksenya-ta) [inline src/app/shared/services/user/authentication.service.ts]: services should not be used that way, it is supposed to be injected as dependency

File: src/app/shared/services/localStorage/local-storage.service.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LocalStorageService {
+
+  constructor() { }
+
+  static setItem(key: string, value: string): void {
+    localStorage.setItem(key, value);
```
Comment (ksenya-ta) [inline src/app/shared/services/localStorage/local-storage.service.ts]: do not use anything from window object directly, create injectable token for windows object (or use any lib that provides it) and add it as a dependency to the services that needs it

File: src/app/shared/services/localStorage/local-storage.service.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  static getItem(key: string): string {
```
Comment (ksenya-ta) [inline src/app/shared/services/localStorage/local-storage.service.ts]: no static methods like this, and even for static methods it should have visibility modifiers set as we agreed before

**Pull request closed without merging**